lib/frontend.ml

```diff
@@ -163,7 +164,12 @@ module Make(C: S.CONFIGURATION with type 'a io = 'a Lwt.t) = struct
       let pages = Io_page.pages num in
       List.iter
         (fun (gref, page) ->
-           let id = gref mod (1 lsl 16) in
+           let rec next () =
+             let id = nf.rx_id in
+             nf.rx_id <- (succ nf.rx_id) mod (1 lsl 16) ;
+             if Hashtbl.mem nf.rx_map id then next () else id
```

Comment:
I agree with minimising changes. Can you just increase the initial value of `Hashtbl.create` then?